### DISCUSSION TITLE
Hide fully booked events but not the list

### DISCUSSION THREAD

**Discussion by Joostvab** [liedekef/events-made-easy#199] [General]
### Discussed in https://github.com/liedekef/events-made-easy/discussions/197

<div type='discussions-op-text'>

<sup>Originally posted by **Joostvab** June 10, 2023</sup>
Dear reader,

In the EME settings (RSVP) it's possible to hide fully booked events? 
But also they disappear from create list [eme_events category=2 scope=this_month template_id=6]. 
While the event is yet to take place so the attendance list is necessary and the event has not to occur  in the calendar.

Is there a way to solve this?

Thanks in advance</div>

Bedankt voor het antwoord maar met de kalender functie kom ik er helaas niet uit.
Doelstelling is dat de kalender alleen events laten zien waarop nog boekingen gedaan kunnen worden.
Wat er is gebruikt:
Events setup

[eme_events category=2 template_ID=6]

Template = 6
[eme_if tag="#_AVAILABLESPACES" value='0']
#_EVENTNAME
Datum: #_STARTDATE
#_ATTENDEES
[/eme_if]

Calender setup

[eme_calendar full=0  htmldiv=0 weekdays=6,0 template_ID=10]

Template = 10
[eme_if tag="#_AVAILABLESPACES" value='1']
category=2
[/eme_if]

wat doe ik in deze verkeerd?

**Comment by liedekef** [comment]
Since this is not a bug, please don't open an issue for this.
Several things here:
- you are talking in Dutch
- you don't mention what is going wrong
- #_AVAILABLESPACES returns the number of free seats, your second check compares that to the value of 1 (so only 1 seat) ==> use the "gt" check (greater then) to check if the number is bigger than 0

**Reply by Joostvab** [reply]
My apologies for the wrong language and that I don't mentioned what is going wrong.

Here's what goes wrong:

two events in one day (A & B).
A = fully booked.
B = is not fully booked.

In the calendar view, I would like to have only event B come back.
In the EME settings (RSVP) it's possible to hide fully booked events but then they are not visible in the lists and this is not what I want.
The bookings from event A had to be show in several lists.

Therefore I'd like to use a calendar code (below) to solve this problem but when I used this, the calendar returns two event A & B.

My question is it possible with a code to show only event B and what mistake did I make?  

code:
[eme_calendar full=0  htmldiv=0  weekdays=6,0 category=2 template_ID=10]

template 10=
[eme_if tag="#_AVAILABLESPACES" gt=0]
#_Eventname
[/eme_if]

**Reply by liedekef** [reply]
if the template doesn't seem to be applied: it is template_id (all lowercase). And as specified in the doc: don’t forget to use `<li>` and `</li>` in the template (in the calendar, it is a ul-list so the entries need the li-element as enclosure:
```
[eme_if tag="#_AVAILABLESPACES" gt=0]
<li>#_Eventname</li>
[/eme_if]
```

**Reply by Joostvab** [reply]
Thank you for your respons and yes you sead so: don’t forget to use <li> and </li> in the template!

However, with the solutions given, the solution does not yet appear to be the solution. The calendar view continues to show the two events even though bookings have been made on them. Both events should therefore not be visible, but unfortunately they are!

I've now added screenshots of everything:
1. Introduction morning event
2. Introduction afternoon event
3.Template 10
4. HTML code
5. Calendar view
6. Introductory reservation not possible

It looks like the template doesn't seem to be applied, even with the right adjustments.

Am I missing something or are there other solutions?
![4  HTML code](https://github.com/liedekef/events-made-easy/assets/128549102/9223f3e3-6cd5-4877-bb51-6d989d569cb5)
![3  Template 10](https://github.com/liedekef/events-made-easy/assets/128549102/c57c4eed-50f3-4013-879c-c93d1b2eb4fd)
![1  Kennismaking ochtend](https://github.com/liedekef/events-made-easy/assets/128549102/d468a345-cf2d-40af-a740-39b7345abbff)
![2  Kennismaking middag](https://github.com/liedekef/events-made-easy/assets/128549102/5ae64468-ff0d-443d-936e-df1943083681)
![6  Kennismaking reserveren niet mogelijk](https://github.com/liedekef/events-made-easy/assets/128549102/fbfbcfce-8f80-4ed6-811d-9b5609e4058e)
![5  Kalendar weergave](https://github.com/liedekef/events-made-easy/assets/128549102/5f69100c-0099-49f1-ba51-0092a1d612b9)

**Reply by liedekef** [reply]
Well, your screenshot for the eme_calendar shortcode shows you're not using the template_id option in it ...

**Comment by Joostvab** [comment]
I can understand your reaction but that was a wrong screenshot the correct one is hereby given. 
Still with all the right code the template doesn't seem to be applied,

Is there are another solution
![The correct Code](https://github.com/liedekef/events-made-easy/assets/128549102/98eba5ca-20ce-4f9e-ac66-ac428209ff35)
s?

**Reply by liedekef** [reply]
Wait ... you're using full=0 ... I need to check the combination of full=0 and template_id

**Reply by liedekef** [reply]
Ok, you're showing a screenshot with "mouse over" active showing the event titles. For the short calendar, that is the only thing currently in place. The template_id is being used for the full calendar only.
If you want to change the "small" calendar layout, you can only do that by changing the generic EME option called "Small calendar title". This is in the EME options, tab "Calendar". There it is also possible to use the eme_if shortcode, but li-enclosing tags are not needed (the event titles are separated by whatever you specify in the option "Small calendar title separator" (which is ", " by default).

**Comment by Joostvab** [comment]
Oke, I used to eme_if code in the generic EME option called "Small calendar title" but still the output was both booked events and is not my intension. The aim is that fully booked events would disappear.

This gave me the idea of using the large calendar and the display of the number of available seats. With any explanation, this makes it clear to visitors that they do not have to select this option. I would have preferred the first solution, but this is also acceptable. The images have been added

Thank you for the support and thinking along. Is greatly appreciated!
![calendar vieuw](https://github.com/liedekef/events-made-easy/assets/128549102/546e77c3-fe77-4379-a306-65a3fedf485f)
![Settings opmaak full calendar](https://github.com/liedekef/events-made-easy/assets/128549102/41cdc31f-2753-44af-8eaf-db52d9dc1558)
![calendar code](https://github.com/liedekef/events-made-easy/assets/128549102/126a15ea-2cba-440c-ac69-289cba15f43e)

**Reply by liedekef** [reply]
Did you try the full calendar with the template? I'll test later on both solutions

**Reply by Joostvab** [reply]
I tried it bit I didn't get the right solution

**Reply by liedekef** [reply]
I just tried for the fullcalendar format:
`[eme_if tag="#_AVAILABLESPACES" gt=0]<li>#_LINKEDNAME</li>[/eme_if]`
and for the small calendar format
`[eme_if tag="#_AVAILABLESPACES" gt=0]#_EVENTNAME[/eme_if]`

and both work as expected.

**Comment by Joostvab** [comment]
Yes it did,
Very happy with It. Thanks very much dhr. Fiedekef
